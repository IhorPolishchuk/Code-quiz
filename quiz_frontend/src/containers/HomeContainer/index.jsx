import React, { Component } from 'react';

import { TextField } from'./../../components/ui';

class HomeContainer extends Component {
    render () {
        return (
            <div>
                <TextField
                    label="text"
                    error={false}
                    regex={/\.com/g}
                    errorMessage="Should contain '.com'"
                />
            </div>
        );
    }
}

export default HomeContainer;
