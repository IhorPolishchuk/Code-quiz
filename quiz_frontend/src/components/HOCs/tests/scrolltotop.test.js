import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ScrollToTopComponent } from './../index.js';
import HomeContainer from './../../../containers/HomeContainer';

describe('ScrollToTop HOC test', () => {
    it('ScrollToTop HOC without children', () => {
        const sttc = shallow(
            <ScrollToTopComponent/>
        );
        const tree = toJson(sttc);

        expect(tree).toMatchSnapshot();
    });
    it('ScrollToTop HOC with children', () => {
        const sttc = shallow(
            <ScrollToTopComponent>
                <HomeContainer/>
            </ScrollToTopComponent>
        );
        const tree = toJson(sttc);

        expect(sttc.find(<HomeContainer/>)).toBeTruthy();
        expect(tree).toMatchSnapshot();
    });
});
