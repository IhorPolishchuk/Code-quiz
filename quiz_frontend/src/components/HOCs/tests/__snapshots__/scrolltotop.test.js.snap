// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop HOC test ScrollToTop HOC with children 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`ScrollToTop HOC test ScrollToTop HOC without children 1`] = `
<Route
  render={[Function]}
/>
`;
